components/overflow-group/overflow-group-mixin.js

```diff
@@ -0,0 +1,349 @@
+			 */
+			autoShow: {
+				type: Boolean,
+				attribute: 'auto-show',
+			},
+			/**
+			 * @ignore
+			 */
+			chompIndex: {
+				type: Number,
+			},
+			/**
+			 * minimum amount of buttons to show
+			 * @type {number}
+			 */
+			minToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'min-to-show',
+			},
+			/**
+			 * maximum amount of buttons to show
+			 * @type {number}
+			 */
+			maxToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-to-show',
+			},
+			/**
+			 * @ignore
+			 */
+			mini: {
+				type: Boolean,
+				reflect: true
+			},
+			/**
+			 * Set the opener type to 'icon' for a `...` menu icon instead of `More actions` text
+			 * @type {'default'|'icon'}
+			 */
+			openerType: {
+				type: String,
+				attribute: 'opener-type'
+			}
+		};
+	}
+
+	static get styles() {
+		return [offscreenStyles, css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-overflow-group-container {
+				display: flex;
+				flex-wrap: wrap;
+				justify-content: var(--d2l-overflow-group-justify-content, normal);
+			}
+			.d2l-overflow-group-container ::slotted([data-is-chomped]) {
+				display: none !important;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._handleItemMutation = this._handleItemMutation.bind(this);
+		this._handleResize = this._handleResize.bind(this);
+
+		this._throttledResize = (entries) => requestAnimationFrame(() => this._handleResize(entries));
+
+		this._overflowHidden = false;
+		this.autoShow = false;
+		this.maxToShow = -1;
+		this.minToShow = 1;
+		this.mini = this.openerType === OPENER_TYPE.ICON;
+		this.openerType = OPENER_TYPE.DEFAULT;
+		this._resizeObserver = null;
+		this._slotItems = [];
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._resizeObserver) this._resizeObserver.disconnect();
+	}
+
+	async firstUpdated() {
+		super.firstUpdated();
+
+		// selected elements
+		this._buttonSlot = this.shadowRoot.querySelector('slot');
+
+		this._container = this.shadowRoot.querySelector('.d2l-overflow-group-container');
+
+		this._resizeObserver = new ResizeObserver(this._throttledResize);
+		this._resizeObserver.observe(this._container);
+	}
+
+	render() {
+		const overflowMenu = this.getOverflowMenu();
+
+		this._slotItems.forEach((element, index) => {
+			if (!this.overflowMenuHidden && index >= this.chompIndex) {
+				element.setAttribute('data-is-chomped', '');
+			} else {
+				element.removeAttribute('data-is-chomped');
+			}
+		});
+
+		return html`
+			<div class="d2l-overflow-group-container">
+				<slot @slotchange="${this._handleSlotChange}"></slot>
+				${overflowMenu}
+			</div>
+		`;
+	}
+
+	update(changedProperties) {
+		super.update(changedProperties);
+
+		if (changedProperties.get('autoShow')) {
+			this._getItemLayouts(this._slotItems);
+			this._autoDetectBoundaries(this._itemLayouts);
+		}
+
+		if (changedProperties.get('minToShow')
+			|| changedProperties.get('maxToShow')) {
```

Comment:
Is the intent here to re-run `_chomp()` if `minToShow` or `maxToShow` changed, or if they changed to a value > 0? `.get` returns the value, whereas `.has` returns whether it's in the collection.

Reply:
This is just the original logic from the overflow group component so I don't know offhand what the intention was, but my guess would be that we would want to run `_chomp` when either of them changed and so `.has` would make more sense.

Reply:
Agreed... I think `.has` makes more sense. Note though that this would have been ignoring the case where the value changed from `undefined`/`null`/`0`, since `get` is returning the old value. Once these values are set, they probably never change in practice, and the initial chomping must have been invoked through another path - maybe that's why this hasn't been an issue.